shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -104,20 +105,38 @@
    *
    * <p>This must be called before any other native methods, and can be overridden by tests to avoid
    * loading native libraries.
+   *
+   * <p>This method should only be called once across all FlutterJNI instances.
    */
   public void loadLibrary() {
+    if (loadLibraryCalled) {
```

Comment:
A poor man's compromise for not splitting this class into a static stateless and stateful component.